utils/fragment_range.hh

```diff
@@ -346,7 +347,7 @@ T read_simple_native(View& v) {
     if (v.current_fragment().size() >= sizeof(T)) [[likely]] {
         auto p = v.current_fragment().data();
         v.remove_prefix(sizeof(T));
-        return *reinterpret_cast<const net::packed<T>*>(p);
+        return read_trivial<T>(p);
```

Comment:
I don't like the name "read_trivial". When seeing this code I don't understand what this is trying to do, and what is "trivial" about it.
How about the name unaligned_read()?
Or, why can't we use bit_cast<T> directly instead of also adding this read_trivial?

Reply:
bit_cast cast values: for example `unsigned long` to `double`, which is done in sstables.cc. read_trivial reads a value from a certain address. You could reimplement bit_cast in terms of read_trivial, but not the other way around.
As for the name, I borrowed the idea from `linearizing_input_stream` which has a similar method with that name. It's "trivial", because it reads trivial objects (https://en.cppreference.com/w/cpp/types/is_trivial) from memory.
But `unaligned_read` is good too. I will change the name.